src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginCache.cs

```diff
@@ -46,6 +47,8 @@ public class TestPluginCache
         /// </summary>
         private AssemblyResolver assemblyResolver;
 
+        private IAssemblyResolver platformAssemblyResolver;
```

Comment:
cannot create in .ctor, since on creation of this object we immediately assign a handler to current domains AssemblyResolve event.

The order of assigning event handler matters since the call to them go in that order itself